packages/app-store/salesforce/lib/CrmService.ts

```diff
@@ -899,28 +907,77 @@ export default class SalesforceCRMService implements CRM {
 
     return writeOnRecordBody;
   }
-  private getTextFieldValue({
+  private async getTextFieldValue({
     fieldValue,
     fieldLength,
     calEventResponses,
+    bookingUid,
   }: {
     fieldValue: string;
     fieldLength: number;
     calEventResponses?: CalEventResponses | null;
+    bookingUid?: string | null;
   }) {
-    let valueToWrite = fieldValue.substring(0, fieldLength);
-    if (!calEventResponses) return valueToWrite;
+    // If no {} then indicates we're passing a static value
+    if (!fieldValue.startsWith("{") && !fieldValue.endsWith("}")) return fieldValue;
 
-    // Check if we need to replace any values with values from the booking questions
-    const regexValueToReplace = /\{(.*?)\}/g;
-    valueToWrite = valueToWrite.replace(regexValueToReplace, (match, captured) => {
-      return calEventResponses[captured]?.value ? calEventResponses[captured].value.toString() : match;
-    });
+    let valueToWrite = fieldValue;
+
+    if (fieldValue.startsWith("{form:")) {
```

Comment:
If we're looking for a routing form response the value needs to be `{form:FieldValue}` vs if it's from a booking response it should be `{fieldValue}`